### DISCUSSION TITLE
.env file confusion.

### DISCUSSION THREAD

**Discussion by XPAULUSX** [nuxt/nuxt#16287] [General]
Hello again,

I'm still trying to figure out how to configure my application for different environments and appear to be running into a few more issues here:

First of all, reading the runtime config docs here:

https://v3.nuxtjs.org/guide/going-further/runtime-config/

It looks as if I should be able to specify values in a .env file like this:

![image](https://user-images.githubusercontent.com/47896035/197785824-b2c362c3-7116-411c-95fc-cccd572e98df.png)

This should allow values within my nuxt.config.ts to be overwritten by these .env file values, however, the docs specify that "you must have a fallback value (which can just be an empty string) defined in your nuxt.config", like this:

![image](https://user-images.githubusercontent.com/47896035/197786353-b8c3eddf-7905-40ba-94d5-1bf3d79f9f7e.png)

This seems to work absolutely perfectly if i'm running "npm run dev", see below:

![image](https://user-images.githubusercontent.com/47896035/197787110-21413aa5-115c-4c69-b2a5-dad2f69086d6.png)

However when I run "npm run build" and then preview the build using "node .output/server/index.mjs" the values are empty, its as if nothing is being pulled from the .env file into the runtime config:

![image](https://user-images.githubusercontent.com/47896035/197787816-7bd319e6-937d-45a9-bc8d-2a53d4dad84a.png)

At this point i tried to take a different approach and pull the env file values into the runtime config manually:

![image](https://user-images.githubusercontent.com/47896035/197788490-f625f3f1-a47e-4455-9ddb-e33952377bce.png)

I have a .env file and a .env.production file, my thought process is that when I run "npm run dev" the .env file values should be passed in, and when i run "npm run build" the .env.production values should be pulled in, however, both scenarios only ever pull in the .env file values.

![image](https://user-images.githubusercontent.com/47896035/197788921-b0640272-d384-4d1a-9473-a9580094587c.png)

I really need to be able to switch per environment, any help with this issue would be greatly appreciated, thank you.

**Comment by danielroe** [comment]
You need to source the environment file yourself in production. The way to do that is different depending on the platform.

On Linux you can simply run `source .env.production && node .output/server/index.mjs`.

**Reply by XPAULUSX** [reply]
That's perfect! our staging/production environments run on Linux!

Any ideas how to do this on Windows too?  This is just so I can demonstrate my learnings to other members of my team.

Thank you for your time and help!

**Reply by alexiokay** [reply]
I used git shell on windows and run command `source ./.production.env && node .output/server/index.mjs`

Then I've got this `Listening http://[::]:3000`

But still it reads only from ,env file in production
